src/blog/2024/11/why-pub-sub-in-uns.md

```diff
@@ -0,0 +1,92 @@
+date: 2024-11-13
+authors: ["sumit-shinde"]
+image: 
+keywords: unified namespace, pub/sub, mqtt, iiot, manufacturing, real-time data, event-driven, scalability, smart factory, flowfuse, iot integration
+tags:
+   - flowfuse
+   - mqtt 
+   - unified namespace
+---
+
+As the manufacturing industry evolves and becomes increasingly connected through the Industrial Internet of Things, the concept of a Unified Namespace (UNS) has emerged as a critical architecture for centralizing and organizing data. UNS serves as a central reference point where all operational data—from machines to sensors—can be accessed in a consistent and structured way. Over time, more and more manufacturers have adopted UNS to simplify data integration and improve real-time visibility across systems.
+
+If you're unfamiliar with UNS, please read our [Introduction to Unified Namespace](/blog/2023/12/introduction-to-unified-namespace/) for a Basic understanding of UNS.
+
+<!--more-->
+
+If you're familiar with or already using a UNS, you might be asking: **Why does a UNS need Pub/Sub**? Here, we’ll explore how combining the Pub/Sub model with a Unified Namespace can help manufacturers streamline data flow, reduce latency, and enable more responsive and scalable operations.
+
+## What is Pub/Sub?
+
+Before discussing why Publish-Subscribe (Pub/Sub) is essential for a UNS, let's first define the Pub/Sub model.
+
+![Publish Subscribe Model](./images/pub-sub.png){data-zoomable}
+_Publish Subscribe Model_
+
+The Pub-Sub model is a way for systems to communicate where one component, called the publisher, sends messages to a central system ( Broker such [MQTT](/blog/2024/06/how-to-use-mqtt-in-node-red/), [RabitMQ](/node-red/protocol/amqp/), and [Kafka](/blog/2024/03/using-kafka-with-node-red/) ), and other components, called subscribers, receive those messages. The publisher doesn’t need to know who the subscribers are, and the subscribers don’t know who the publishers are. The central system, or broker, ensures the right messages go to the right subscribers based on their interests.
+
+## Why Does a UNS Need Pub/Sub?
+
+Now that you have a basic understanding of Pub/Sub, let’s dive into why this architecture is essential for a Unified Namespace.
+
+### Decoupling Producers and Consumers with Flexible Communication
+
+In a typical manufacturing setup, various systems and machines operate in silos. Sensors, control systems, maintenance platforms, and quality control software often struggle to share data. This leads to inefficiencies and delays in accessing the right data when it’s needed most. Traditional approaches—such as point-to-point integrations—often become cumbersome as more devices and systems are added to the factory floor.
```

Comment:
Please also explain that the point to point system blocks innovation and link to: https://flowfuse.com/blog/2023/08/isa-95-automation-pyramid-to-unified-namespace/